cub/test/catch2_test_nvrtc.cu

```diff
@@ -0,0 +1,307 @@
+    "    if (sum != (threads_per_block * (threads_per_block - 1) / 2))                           \n"
+    "    {                                                                                       \n"
+    "      atomicAdd(errors, 1);                                                                 \n"
+    "    }                                                                                       \n"
+    "  }                                                                                         \n"
+    "                                                                                            \n"
+    "  // Test block scan                                                                        \n"
+    "  int prefix_sum{};                                                                         \n"
+    "  block_scan_t(block_scan_storage).InclusiveSum(items[0], prefix_sum);                      \n"
+    "  if (prefix_sum != (threadIdx.x * (threadIdx.x + 1) / 2))                                  \n"
+    "  {                                                                                         \n"
+    "    atomicAdd(errors, 1);                                                                   \n"
+    "  }                                                                                         \n"
+    "                                                                                            \n"
+    "  // Test block radix sort                                                                  \n"
+    "  block_radix_sort_t(block_radix_sort_storage).SortDescending(items);                       \n"
+    "                                                                                            \n"
+    "  // Test block store                                                                       \n"
+    "  block_store_t(block_store_storage).Store(ptr, items);                                     \n"
+    "}                                                                                           \n";
+
+  const char* name = "test";
+
+  REQUIRE(NVRTC_SUCCESS == nvrtcCreateProgram(&prog, src, name, 0, nullptr, nullptr));
+
+  constexpr int num_includes = 4;
+
+  const char* includes[num_includes] = {
+    NVRTC_CUB_PATH,
+    NVRTC_THRUST_PATH,
+    NVRTC_LIBCUDACXX_PATH,
+    NVRTC_CTK_PATH
+  };
+
+  std::size_t log_size{};
+  nvrtcResult compile_result = nvrtcCompileProgram(prog, num_includes, includes);
+  
+  REQUIRE(NVRTC_SUCCESS == nvrtcGetProgramLogSize(prog, &log_size));
+
+  std::unique_ptr<char[]> log{ new char[log_size] };
+  REQUIRE(NVRTC_SUCCESS == nvrtcGetProgramLog(prog, log.get()));
+  INFO("nvrtc log = " << log.get());
+  REQUIRE(NVRTC_SUCCESS == compile_result);
+
+  std::size_t ptx_size{};
+  REQUIRE(NVRTC_SUCCESS == nvrtcGetPTXSize(prog, &ptx_size));
+
+  std::unique_ptr<char[]> ptx{ new char[ptx_size] };
+  REQUIRE(NVRTC_SUCCESS == nvrtcGetPTX(prog, ptx.get()));
```

Comment:
We're going to run into problems by relying on PTX JIT because our GPU runners don't always have the latest driver, see https://github.com/NVIDIA/cccl/issues/268